src/VariationPriceHandler.php

```diff
@@ -0,0 +1,211 @@
+      $product_variation->setPrice($newPrice);
+      $product_variation->save();
+    }
+    return TRUE;
+  }
+
+  /**
+   * Retrieve all product variation types associated with the attribute.
+   *
+   * @param Drupal\commerce_attributes\Entity\ProductAttribute $productAttribute
+   *   The ProductAttribute Entity, which is being updated.
+   *
+   * @return array
+   *   Array of all product variations types associated with the attribute.
+   */
+  public function getAllVariationTypes(ProductAttribute $productAttribute) {
+    $types = [];
+    $usedAttributes = $this->getAllUsedAttributes();
+    foreach ($usedAttributes as $variation_type_id => $attributes) {
+      if (in_array($productAttribute->id(), $attributes)) {
+        $types[] = $variation_type_id;
+      }
+    }
+    return $types;
+  }
+
+  /**
+   * Retrieves all attributes used by product variation types.
+   *
+   * @return array
+   *   Associative array which has product variation type as key and
+   *   the attributes used as value.
+   */
+  public function getAllUsedAttributes() {
+    $usedAttributes = [];
+    $types = [];
+    $variation_type_storage = $this->entityTypeManager->getStorage('commerce_product_variation_type');
+    $variation_types = $variation_type_storage->loadMultiple();
+    $attribute_field_map = $this->attributeFieldManager->getFieldMap();
+    foreach ($variation_types as $variation_type_id => $variation_type) {
+      $usedAttributes[$variation_type_id] = array_column($attribute_field_map[$variation_type_id], 'attribute_id');
+    }
+    return $usedAttributes;
+  }
+
+  /**
+   * Provides the total price of attribues attached to a variation.
+   *
+   * @param string $variation_type_id
+   *   The variation type machine name.
+   * @param Drupal\commerce_product\Entity\ProductVariation $variation
+   *   The commerce product variation.
+   *
+   * @return int
+   *   Sum of all attribute values associated with a variation.
+   */
+  public function getAttributePriceSum($variation_type_id, ProductVariation $variation) {
+    $attributeTotalPrice = 0;
+    $usedAttributes = $this->getAllUsedAttributes();
+    $variationTypeAttributes = $usedAttributes[$variation_type_id];
+    foreach ($variationTypeAttributes as $attribute) {
+      // The attribute field name of variation type will always be in the
+      // format of `attrubute_attribute_machine_name`.
+      $attributefieldName = 'attribute_' . $attribute;
+      $attributeValueID = $variation->get($attributefieldName)->getValue()['0']['target_id'];
+      $productAttributeValue = $this->entityManager->getStorage('commerce_product_attribute_value')->load($attributeValueID);
+      if (!empty($productAttributeValue->get('price')->getValue()['0']['number'])) {
+        $attributePrice = $productAttributeValue->get('price')->getValue()['0']['number'];
+        $attributeTotalPrice += $attributePrice;
+      }
+    }
+    return $attributeTotalPrice;
```

Comment:
**CHANGE REQUEST**

See above, use the Price API to do maths and return a Price object.

Reply:
Done